worlds/cv64/rom.py

```diff
@@ -0,0 +1,907 @@
+import Utils
+
+from worlds.Files import APDeltaPatch
+
+import hashlib
+import os
+import pkgutil
+
+from . import lzkn64
+from .data import patches
+from .stages import get_stage_info
+from .text import cv64_string_to_bytes, cv64_text_truncate, cv64_text_wrap
+from .aesthetics import renon_item_dialogue, get_item_text_color
+from .options import CV64Options
+from .locations import get_location_info
+from settings import get_settings
+
+CV64US10HASH = '1cc5cf3b4d29d8c3ade957648b529dc1'
+ROM_PLAYER_LIMIT = 65535
+
+warp_map_offsets = [0xADF67, 0xADF77, 0xADF87, 0xADF97, 0xADFA7, 0xADFBB, 0xADFCB, 0xADFDF]
+
+
+class LocalRom(object):
+
+    def __init__(self, file):
+        self.orig_buffer = None
+
+        with open(file, 'rb') as stream:
+            self.buffer = bytearray(stream.read())
+
+    def read_bit(self, address: int, bit_number: int) -> bool:
+        bitflag = (1 << bit_number)
+        return (self.buffer[address] & bitflag) != 0
+
+    def read_byte(self, address: int) -> int:
+        return self.buffer[address]
+
+    def read_bytes(self, startaddress: int, length: int) -> bytes:
+        return self.buffer[startaddress:startaddress + length]
+
+    def write_byte(self, address: int, value: int):
+        self.buffer[address] = value
+
+    def write_bytes(self, startaddress: int, values: list):
+        self.buffer[startaddress:startaddress + len(values)] = values
+
+    def write_int16(self, address, value: int):
+        value = value & 0xFFFF
+        self.write_bytes(address, [(value >> 8) & 0xFF, value & 0xFF])
+
+    def write_int16s(self, startaddress, values: list):
+        for i, value in enumerate(values):
+            self.write_int16(startaddress + (i * 2), value)
+
+    def write_int24(self, address, value: int):
+        value = value & 0xFFFFFF
+        self.write_bytes(address, [(value >> 16) & 0xFF, (value >> 8) & 0xFF, value & 0xFF])
+
+    def write_int24s(self, startaddress, values: list):
+        for i, value in enumerate(values):
+            self.write_int24(startaddress + (i * 3), value)
+
+    def write_int32(self, address, value: int):
+        value = value & 0xFFFFFFFF
+        self.write_bytes(address, [(value >> 24) & 0xFF, (value >> 16) & 0xFF, (value >> 8) & 0xFF, value & 0xFF])
+
+    def write_int32s(self, startaddress, values: list):
+        for i, value in enumerate(values):
+            self.write_int32(startaddress + (i * 4), value)
+
+    def write_to_file(self, file):
+        with open(file, 'wb') as outfile:
+            outfile.write(self.buffer)
+
+    def read_from_file(self, file):
+        with open(file, 'rb') as stream:
+            self.buffer = bytearray(stream.read())
+
+
+def patch_rom(multiworld, options: CV64Options, rom, player, offset_data, active_stage_exits, s1s_per_warp,
+              active_warp_list, required_s2s, total_s2s, shop_name_list, shop_desc_list, shop_colors_list, slot_name,
+              active_locations):
```

Comment:
As long as it might make this function definition, type hints would be very useful.